bench/clblast_benchmark.cpp

```diff
@@ -0,0 +1,333 @@
+        clReleaseEvent(events[0]);
+        clReleaseEvent(events[1]);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(scal3op_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT alpha(2.4367453465);
+      MemBuffer<ScalarT> buf1(context, size);
+      MemBuffer<ScalarT> buf2(context, size);
+      MemBuffer<ScalarT> buf3(context, size);
+
+      cl_event events[3];
+      flops = benchmark<>::measure(no_reps, size * 3, [&]() {
+        clblast::Scal<ScalarT>(size, alpha, buf1.dev(), 0, 1, context._queue(),
+                               &events[0]);
+        clblast::Scal<ScalarT>(size, alpha, buf2.dev(), 0, 1, context._queue(),
+                               &events[1]);
+        clblast::Scal<ScalarT>(size, alpha, buf3.dev(), 0, 1, context._queue(),
+                               &events[2]);
+        clWaitForEvents(3, events);
+        clReleaseEvent(events[0]);
+        clReleaseEvent(events[1]);
+        clReleaseEvent(events[2]);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(axpy3op_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT alphas[] = {1.78426458744, 2.187346575843, 3.78164387328};
+      size_t offsets[] = {0, size, size * 2};
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufsrc(context, size * 3);
+      MemBuffer<ScalarT> bufdst(context, size * 3);
+
+      cl_event event;
+      flops = benchmark<>::measure(no_reps, size * 3 * 2, [&]() {
+        clblast::AxpyBatched<ScalarT>(size, alphas, bufsrc.dev(), offsets, 1,
+                                      bufdst.dev(), offsets, 1, 3,
+                                      context._queue(), &event);
+      });
+      clWaitForEvents(1, &event);
+      clReleaseEvent(event);
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(blas1_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT alpha(3.135345123);
+      MemBuffer<ScalarT> buf1(context, size);
+      MemBuffer<ScalarT> buf2(context, size);
+      ScalarT vr[3];
+      size_t vi;
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, vr, 3);
+      MemBuffer<size_t, CL_MEM_READ_ONLY> buf_i(context, &vi, 1);
+
+      cl_event events[5];
+      flops = benchmark<>::measure(no_reps, size * 12, [&]() {
+        clblast::Axpy<ScalarT>(size, alpha, buf1.dev(), 0, 1, buf2.dev(), 0, 1,
+                               context._queue(), &events[0]);
+        clblast::Asum<ScalarT>(size, bufr.dev(), 0, buf2.dev(), 0, 1,
+                               context._queue(), &events[1]);
+        clblast::Dot<ScalarT>(size, bufr.dev(), 1, buf1.dev(), 0, 1, buf2.dev(),
+                              0, 1, context._queue(), &events[2]);
+        clblast::Nrm2<ScalarT>(size, bufr.dev(), 2, buf1.dev(), 0, 1,
+                               context._queue(), &events[3]);
+        clblast::Amax<ScalarT>(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                               context._queue(), &events[4]);
+        clWaitForEvents(5, events);
```

Comment:
I am not sure it is a part of the standard. There is an example of how to use sizeof(block), but I have not found an explicit statement.